lib/pbench/cli/server/report.py

```diff
@@ -158,56 +164,81 @@ def report_cache(tree: CacheManager):
+            cached_count += 1
             try:
-                referenced = tarball.last_ref.stat().st_mtime
+                referenced = (cache / "last_ref").stat().st_mtime
             except Exception as e:
-                detailer.error(f"{tarball.name} last ref access: {str(e)!r}")
+                detailer.error(f"{dsname} last ref access: {str(e)!r}")
                 last_ref_errors += 1
             else:
-                agecomp.add(tarball.name, referenced)
-            if tarball.unpacked:
-                cached_count += 1
-        size = Metadata.getvalue(tarball.dataset, Metadata.SERVER_UNPACKED)
+                agecomp.add(dsname, referenced)
         if not size:
-            detailer.error(f"{tarball.name} has no unpacked size")
+            detailer.error(f"{dsname} has no unpacked size")
             lacks_size += 1
         elif not isinstance(size, int):
             detailer.error(
-                f"{tarball.name} has non-integer unpacked size "
+                f"{dsname} has non-integer unpacked size "
                 f"{size!r} ({type(size).__name__})"
             )
             bad_size += 1
         else:
-            sizecomp.add(tarball.name, size)
+            sizecomp.add(dsname, size)
             cached_size += size
 
             # Check compression ratios
-            tar_size = tarball.tarball_path.stat().st_size
-            ratio = float(size - tar_size) / float(size)
-            compcomp.add(tarball.name, ratio)
-        metrics = Metadata.getvalue(tarball.dataset, Metadata.SERVER_UNPACK_PERF)
+            if tarball:
+                tar_size = tarball.stat().st_size
+                ratio = float(size - tar_size) / float(size)
+                compcomp.add(dsname, ratio)
         if not metrics:
-            detailer.message(f"{tarball.name} has no unpack metrics")
+            # NOTE: message not error since nobody has this yet (noise)
+            detailer.message(f"{dsname} has no unpack metrics")
             lacks_metrics += 1
         elif not isinstance(metrics, dict) or {"min", "max", "count"} - set(
             metrics.keys()
         ):
             detailer.error(
-                f"{tarball.name} has bad unpack metrics "
+                f"{dsname} has bad unpack metrics "
                 f"{metrics!r} ({type(metrics).__name__})"
             )
             bad_metrics += 1
         else:
+            found_metrics = True
             unpacked_count += 1
             unpacked_times += metrics["count"]
-            speedcomp.add(tarball.name, metrics["min"], metrics["max"])
+            speedcomp.add(dsname, metrics["min"], metrics["max"])
         if size and metrics:
             stream_fast = size / metrics["min"] / MEGABYTE_FP
             stream_slow = size / metrics["max"] / MEGABYTE_FP
-            streamcomp.add(tarball.name, stream_slow, stream_fast)
+            streamcomp.add(dsname, stream_slow, stream_fast)
         else:
             stream_unpack_skipped += 1
```

Comment:
I think the `if` statement at line 238 should be in the block where we set `found_metrics` (and then it doesn't need to test `metrics`):  the prints for the `streamcomp` values are already protected by `found_metrics`, so it makes sense to have the accumulators protected the same way; also, the test of `metrics` at line 238 is not as rigorous as the one at line 225 (which feeds into the value of `found_metrics`, so, with the code as it stands, we risk problems if the `"min"` or `"max"` key is missing (which wouldn't be a problem if the code were inside the block at 233).

It does mean that `stream_unpack_skipped` will only count a subset of the bad values, but that's probably OK, given that we have the counts in `lacks_metrics` and `bad_metrics`.

Reply:
I'm going to change it, although I'm not entirely convinced. It makes the meaning of `stream_unpack_skipped` a bit weird since it's only "things that were skipped because they have metrics but not size". Then again, right now nothing has metrics and in theory everything that's ever unpacked will have size. (Meaning this should be a subset of "never unpacked" anyway.)